rust/ql/test/library-tests/dataflow/sources/test_futures_io.rs

```diff
@@ -61,16 +61,16 @@ async fn test_futures_rustls_futures_io() -> io::Result<()> {
         let mut cx = Context::from_waker(futures::task::noop_waker_ref());
         let buffer = pinned.poll_fill_buf(&mut cx);
         if let Poll::Ready(Ok(buf)) = buffer {
-            sink(&buffer); // $ hasTaintFlow=url
+            sink(&buffer); // $ MISSING: hasTaintFlow=url
```

Comment:
Currently the computed canonical path for this function is `<core::pin::Pin as tokio::io::async_buf_read::AsyncBufRead>::poll_fill_buf` which makes very little sense.  I guess we should implement some auto-unpinning or something @hvitved , what do you think?

Reply:
@geoffw0 To make the tests pass we could add an additional model for `<core::pin::Pin as tokio::io::async_buf_read::AsyncBufRead>::poll_fill_buf` and friends or would a model like  `<_ as tokio::io::async_buf_read::AsyncBufRead>::poll_fill_buf` also work? If so that would be preferred.

Reply:
It's passing now, but thanks.

Reply:
Ah nice the `<_ as` works as I hoped. Much better indeed!